src/Adapter/Kpi/ShoppingCartTotalKpi.php

```diff
@@ -31,18 +31,43 @@
 use Group;
 use HelperKpi;
 use Order;
+use PrestaShop\PrestaShop\Adapter\LegacyContext;
 use PrestaShop\PrestaShop\Core\Kpi\KpiInterface;
-use Tools;
+use PrestaShop\PrestaShop\Core\Localization\Locale;
+use PrestaShop\PrestaShop\Core\Localization\Locale\Repository as LocaleRepository;
 use Validate;
 
 /**
  * {@inheritdoc}
  */
 final class ShoppingCartTotalKpi implements KpiInterface
 {
-    /** @var array */
+    /**
+     * @var LegacyContext
+     */
+    private $context;
+
+    /**
+     * @var Locale
+     */
+    private $locale;
+
+    /**
+     * @var array
+     */
     private $options;
 
+    /**
+     * @param LocaleRepository $repository
+     */
+    public function __construct(LocaleRepository $repository)
```

Comment:
Maybe it would better to inject a Locale object directly
Then you probably will need a new method in LocaleRepository, getContextLocale or something (there's probably a better solution)

Reply:
Having such entry in a service would allow you to simplify all the classes you modified here by adding the repo and the LegacyContext And I'm sure we will need this kind of feature in the future

Reply:
:heavy_check_mark: